runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java

```diff
@@ -351,7 +351,15 @@ public void onChatMessage(ChatMessage event)
 
 		if (config.screenshotKills() && KILL_MESSAGES.stream().anyMatch(chatMessage::contains))
 		{
-			String fileName = "Kill " + format(new Date());
+			String kill_name = chatMessage;
+
+			for(String test_string : KILL_MESSAGES) {
```

Comment:
As stated on the other PR and in discord, this is better remade as a regex statement with a group matcher for the username.

Reply:
I am curious what you meant by this before, can you give an example of how that regex string would look? The position of the username in the string changes based on the message.